HW04/Ivan_Papushoi/array_my_methods_spec.rb

```diff
@@ -0,0 +1,58 @@
+# frozen_string_literal: true
+
+require_relative 'ivan_papushoi_hw04.rb'
+require 'rspec/autorun'
+
+RSpec.describe Array do
+  describe '#my_select' do
+    context 'with array and odd? method' do
+      let(:array) { [1, 2, 3, 4, 5, 6] }
+      let(:term) { ->(x) { x.odd? } }
+      let(:expectation) { [1, 3, 5] }
+
+      it { expect(array.my_select(&term)).to eq(expectation) }
+    end
+
+    context 'when selecting negatives from random array' do
+      let(:array) { described_class.new(20) { rand(-100..100) } }
+      let(:term) { ->(x) { x.negative? } }
+
+      it { expect(array.my_select(&term)).to eq(array.select(&term)) }
+    end
+  end
+
+  describe '#my_each' do
+    let(:expectation) { 'value 1 value 2 value 3 ' }
+    let(:term) { ->(x) { print "value #{x} " } }
+    let(:array) { [1, 2, 3] }
+
+    context 'when method printing each value in stdout' do
+      it { expect { array.my_each(&term) }.to output(expectation).to_stdout }
+    end
+
+    context 'when return each value from array' do
+      let(:array) { [1, 2, 3] }
+      let(:term) { ->(x) { x } }
+      let(:expectation) { [1, 2, 3] }
+
+      it { expect(array.my_each(&term)).to eq(expectation) }
+    end
+  end
+
+  describe '#my_map' do
+    context 'when method multiply each value' do
+      let(:array) { [1, 2, 3] }
+      let(:term) { ->(x) { x * 2 } }
+      let(:expectation) { [2, 4, 6] }
+
+      it { expect(array.my_map(&term)).to eq(expectation) }
+    end
+
+    context 'when #my_map transform each string' do
```

Comment:
Please, be consistent with other descriptions
```suggestion
    context 'when method transform each string' do
```